app/src/Service/Ldap/Client.php

```diff
@@ -77,4 +79,77 @@ public function check(
 
         return $result;
     }
+
+    /**
+     * @return Entry[]|\Symfony\Component\Ldap\Adapter\CollectionInterface
+     *
+     * @psalm-return \Symfony\Component\Ldap\Adapter\CollectionInterface|array<array-key, Entry>
+     */
+    public function executeQuery(string $query)
+    {
+        $this->ldap->bind($this->config['search_dn'], $this->config['search_password']);
+        return $this->ldap->query($this->config['base_dn'], $query)->execute();
+    }
+
+    public function create(string $dn, array $attributes): bool
+    {
+        // TODO Do not bind inside search (must be done before)
+        $this->ldap->bind($this->config['search_dn'], $this->config['search_password']);
+
+        $entry = new Entry($dn, $attributes);
+        $entryManager = $this->ldap->getEntryManager();
+
+        return $entryManager->add($entry);
+    }
+
+    /**
+     * @return bool|null
+     */
+    public function update(string $query, array $attributes) : bool
+    {
+        $entryManager = $this->ldap->getEntryManager();
+
+        // Finding and updating an existing entry
+        $result = $this->executeQuery($query);
+
+        // FIXME Check result before doing anything on it
+        $entry = $result[0];
+        if ($entry===null) {
```

Comment:
Not a valid test. If result is an empty array, `$entry` will be `undefined` not `null`. Use `empty` method instead.
```suggestion
        if (empty($entry)) {
```